remote/gitea/gitea.go

```diff
@@ -295,7 +297,44 @@ func (c *client) File(u *model.User, r *model.Repo, b *model.Build, f string) ([
 }
 
 func (c *client) Dir(u *model.User, r *model.Repo, b *model.Build, f string) ([]*remote.FileMeta, error) {
-	return nil, fmt.Errorf("Not implemented")
+	var errors string
+	var configs []*remote.FileMeta
+
+	client, err := c.newClientToken(u.Token)
+	if err != nil {
+		return nil, err
+	}
+
+	// List files in repository. Path from root
+	tree, _, err := client.GetTrees(r.Owner, r.Name, b.Commit, true)
+	if err != nil {
+		return nil, err
+	}
+
+	f = path.Clean(f) // We clean path and remove trailing slash
+	f += "/" + "*"    // construct pattern for match i.e. file in subdir
+	for _, e := range tree.Entries {
+		// Filter path matching pattern and type file (blob)
+		if m, _ := filepath.Match(f, e.Path); m && e.Type == "blob" {
+			data, err := c.File(u, r, b, e.Path)
+			if err != nil {
+				errors += fmt.Sprintf("cannot get %s: %s", e.Path, err)
+			}
+
+			config := remote.FileMeta{
+				Name: e.Path,
+				Data: data,
+			}
+
+			configs = append(configs, &config)
```

Comment:
```suggestion
			configs = append(configs, &remote.FileMeta{
				Name: e.Path,
				Data: data,
			})
```
just a code format suggestion ...